quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -259,9 +277,12 @@ successful r =
     QC.Success {} -> True
     _ -> False
 
-makeReplayMsg :: Int -> Int -> String
-makeReplayMsg seed size = let
-    sizeStr = if (size /= defaultMaxSize)
-                 then printf " --quickcheck-max-size=%d" size
-                 else ""
-  in printf "Use --quickcheck-replay=%d%s to reproduce." seed sizeStr
+makeReplayMsg :: (QCGen, Int) -> String
+makeReplayMsg seedSz =
+  printf "Use --quickcheck-single-replay=\"%s\" to reproduce." (show seedSz)
```

Comment:
Probably we should display both the value of `--quickcheck-replay` and `--quickcheck-single-replay` here. Otherwise, the user of `--quickcheck-replay` has no way to get a seed.

Reply:
What are the remaining use cases for `--quickcheck-replay` now? I guess you can just reuse it without introducing a new command line flag.

Reply:
I don't know who is relying on the old behavior. I'm happy to break that if nobody cares.

Reply:
I find it hard to imagine that someone would consciously prefer old behaviour to a new, but it seems you can support both without too much trouble. If it's `--quickcheck-replay=(QCGen ..., 42)` do a new thing, if it's `--quichckeck-replay=QCGen...` do the old thing.

Reply:
And when printing, should we show both forms when sensible?

Reply:
Showing one form would be enough IMO.

Reply:
It can be done as you suggest. But note that the old form is unusable if the user has no way to discover the seeds that go with it. What would be the purpose of keeping the old form then?

Reply:
Just an escape hatch. I do not see why a user might want the old behavior back; but someone could have a regression script still using an old format. Not crashing in such scenario would be useful.

Reply:
Pushed. Also added a note to the change log. Thanks for your review @Bodigrim!